e2e/ruler_test.go

```diff
@@ -25,174 +25,212 @@ import (
+		}
 	}
 }
 
-func createRuleEvaluatorSecrets(ctx context.Context, t *OperatorContext, certSecretName, tokenSecretName string, cert, key []byte) {
-	secrets := []*corev1.Secret{
-		{
+func testRuleEvaluatorSecrets(ctx context.Context, t *testing.T, kubeClient kubernetes.Interface, opClient versioned.Interface) func(*testing.T) {
+	return func(t *testing.T) {
+		cert, key, err := cert.GenerateSelfSignedCertKey("test", nil, nil)
+		if err != nil {
+			t.Errorf("generating tls cert and key pair: %s", err)
+		}
+
+		tlsPair := &corev1.Secret{
 			ObjectMeta: metav1.ObjectMeta{
-				Name:      tokenSecretName,
-				Namespace: t.pubNamespace,
+				Name:      "alertmanager-authorization",
+				Namespace: operator.DefaultPublicNamespace,
 			},
 			Data: map[string][]byte{
 				"token": []byte("auth-bearer-password"),
 			},
-		},
-		{
+		}
+		authToken := &corev1.Secret{
 			ObjectMeta: metav1.ObjectMeta{
-				Name:      certSecretName,
-				Namespace: t.pubNamespace,
+				Name:      "alertmanager-tls",
+				Namespace: operator.DefaultPublicNamespace,
 			},
 			Data: map[string][]byte{
 				"cert": cert,
 				"key":  key,
 			},
-		},
-	}
-
-	for _, s := range secrets {
-		if err := t.Client().Create(ctx, s); err != nil {
-			t.Fatalf("create alertmanager secret %q: %s", s.Name, err)
 		}
-	}
-}
-
-// createRuleEvaluatorOperatorConfig ensures an OperatorConfig can be deployed
-// that contains rule-evaluator configuration.
-func createRuleEvaluatorOperatorConfig(ctx context.Context, t *OperatorContext, certSecretName, tokenSecretName string) {
-	// Setup TLS secret selectors.
-	certSecret := &corev1.SecretKeySelector{
-		LocalObjectReference: corev1.LocalObjectReference{
-			Name: certSecretName,
-		},
-		Key: "cert",
-	}
 
-	keySecret := certSecret.DeepCopy()
-	keySecret.Key = "key"
-
-	t.createOperatorConfigFrom(ctx, monitoringv1.OperatorConfig{
-		Rules: monitoringv1.RuleEvaluatorSpec{
-			ExternalLabels: map[string]string{
-				"external_key": "external_val",
-			},
-			QueryProjectID: projectID,
-			Alerting: monitoringv1.AlertingSpec{
-				Alertmanagers: []monitoringv1.AlertmanagerEndpoints{
-					{
-						Name:       "test-am",
-						Namespace:  t.userNamespace,
-						Port:       intstr.IntOrString{IntVal: 19093},
-						Timeout:    "30s",
-						APIVersion: "v2",
-						PathPrefix: "/test",
-						Scheme:     "https",
-						Authorization: &monitoringv1.Authorization{
-							Type: "Bearer",
-							Credentials: &corev1.SecretKeySelector{
-								LocalObjectReference: corev1.LocalObjectReference{
-									Name: tokenSecretName,
-								},
-								Key: "token",
-							},
-						},
-						TLS: &monitoringv1.TLSConfig{
-							Cert: &monitoringv1.SecretOrConfigMap{
-								Secret: certSecret,
-							},
-							KeySecret: keySecret,
-						},
-					},
-				},
-			},
-		},
-	})
-}
+		_, err = kubeClient.CoreV1().Secrets(operator.DefaultPublicNamespace).Create(ctx, tlsPair, metav1.CreateOptions{})
```

Comment:
all of it has opportunity for simplification (if _, err := ... ; err != nil ), will stop commenting on this one (:

Reply:
sgtm :)